apps/angularlogos/src/app/search/search.component.spec.ts

```diff
@@ -8,17 +11,63 @@ describe('SearchComponent', () => {
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
+      imports: [NoopAnimationsModule, MatFormFieldModule, MatInputModule, MatIconModule, ReactiveFormsModule],
       declarations: [SearchComponent]
     }).compileComponents();
   }));
 
   beforeEach(() => {
     fixture = TestBed.createComponent(SearchComponent);
     component = fixture.componentInstance;
-    fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should render', () => {
+    expect(fixture).toMatchSnapshot();
+  });
+
+  describe('ngOnChanges', () => {
+    it('should set a new value', () => {
+      // prepare
+      component.searchTerm = 'search-for';
+      jest.spyOn(component.searchControl, 'setValue');
+
+      // act
+      component.ngOnChanges();
+
+      // check
+      expect(component.searchControl.setValue).toHaveBeenCalledWith(component.searchTerm, { emitEvent: false });
+    });
+  });
+
+  describe('clear', () => {
+    const EMPTY_STRING = '';
```

Comment:
Do we really need this abstraction?

Reply:
Does it really hurt? It's a unit test.. don't really see what the downside of it is.

Reply:
It doesn't hurt but I don't see the value of this.

Reply:
I see a point when you give it a very generic name like `expectedValue` because this can be easily replaced should requirements change.

However, those assertions...

```
expect(component.searchControl.setValue).toHaveBeenCalledWith(EMPTY_STRING);
expect(component.searchControl.value).toBe(EMPTY_STRING);
```

have the same value like

```
expect(component.searchControl.setValue).toHaveBeenCalledWith('');
expect(component.searchControl.value).toBe('');
```

The first is not better readable and not better maintainable. IMO even the contrary: It has an abstratcion that you need to parse before you understand it.

Reply:
I'll revert it if it makes you sleep better, don't really see the value of having a discussion over it honestly.

Reply:
Everything's fine 😉 I hoped you could convince me that this is a valuable pattern and I could learn something new!